### PR TITLE
CADENZA-35867 feat: CadenzaJS reload API function

### PR DESCRIPTION
**Pull request by andkoons**

### CODE REVIEW

File: src/docs.md [hunk 1]
```diff
@@ -355,6 +355,16 @@ button.textContent = 'Download Excel';
 button.onclick = () => cadenzaClient.downloadData('{embeddingTargetId}', 'excel');
 ```
 
+### Reload the iframe that is used to show Cadenza
+
+<small>API: [CadenzaClient#reload](./classes/CadenzaClient.html#reload)</small>
+
+Reload the iframe that is used to show Cadenza. This can be used to reload the worksheet and it's views.
```
Comment (jkissel) [inline src/docs.md:362]: it's => its

Also I don't think it's helpful to repeat the heading here.

**Review by jkissel — CHANGES_REQUESTED** [review]

File: sandbox.html [hunk 2]
```diff
@@ -506,6 +508,7 @@
 <template data-action="setSelection" data-common="layer,layer-selection"></template>
 <template data-action="addSelection" data-common="layer,layer-selection"></template>
 <template data-action="removeSelection" data-common="layer,layer-selection"></template>
+<template data-action="reload"><div>No Paramters needed.</div></template>
```
[the remark this answers is not in the document]
Comment (andkoons) [reply sandbox.html]: Right, removed that.

File: src/cadenza.js [hunk 3]
```diff
@@ -801,6 +801,22 @@ export class CadenzaClient {
     this.#download(resolvePath(source), params);
   }
 
+  /**
+   * Reload the iframe that is used for embedding Cadenza.
+   *
+   * @param {AbortSignal} [signal] - A signal to abort the iframe loading
+   * @return {Promise<void>} A `Promise` for when the iframe is loaded
+   */
+  reload(/** @type AbortSignal | undefined */ signal) {
```
[the remark this answers is not in the document]
Comment (andkoons) [reply src/cadenza.js]: (y)

[the remark this answers is not in the document]
Comment (andkoons) [reply src/cadenza.js]: Added the log.

File: src/cadenza.js [hunk 4]
```diff
@@ -801,6 +801,22 @@ export class CadenzaClient {
...
+    if (this.#requiredIframe.contentWindow) {
+      this.#requiredIframe.contentWindow.location.reload();
+      return this.#getIframePromise(signal);
+    }
+    return Promise.reject(
```
[the remark this answers is not in the document]
Comment (andkoons) [reply src/cadenza.js]: Yes, reloading just the views to keep the UI state is better, changed that, also the reload is triggered by an event now.

File: src/cadenza.test.ts [hunk 5]
```diff
@@ -67,6 +67,9 @@ describe('Given a Cadenza JS client instance', () => {
       'present',
     ));
 
+  it('Throws when attempting reload without an iframe', () =>
```
[the remark this answers is not in the document]
Comment (andkoons) [reply src/cadenza.test.ts]: Changed.

File: src/docs.md (hunk 1 above)
Comment (andkoons) [reply src/docs.md]: Changed.

**Pull request merged**